analyzer/src/main/kotlin/managers/Gradle.kt

```diff
@@ -164,6 +171,17 @@ class Gradle(
             gradleConnector.daemonMaxIdleTime(10, TimeUnit.SECONDS)
         }
 
+        // Gradle's default maximum heap is 512 MiB which is too low for bigger projects,
+        // see https://docs.gradle.org/current/userguide/build_environment.html#sec:configuring_jvm_memory.
+        // Set the value to empirically determined 8 GiB if no value is set in "~/.gradle/gradle.properties".
+        val jvmArgs = gradleProperties.find { (key, _) ->
+            key == "org.gradle.jvmargs"
+        }?.second?.split(" ").orEmpty().toMutableList()
+
+        if (jvmArgs.none { it.contains(other = "-xmx", ignoreCase = true) }) {
+            jvmArgs.add("-Xmx8g")
```

Comment:
Please use the more readable `+=` instead of `.add()` (also see my code suggestion below).